apps/www/registry/default/ui/phone-input.tsx

```diff
@@ -0,0 +1,156 @@
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/registry/default/ui/popover"
+
+type PhoneInputProps = Omit<
+  React.InputHTMLAttributes<HTMLInputElement>,
+  "onChange" | "value"
+> &
+  Omit<RPNInput.Props<typeof RPNInput.default>, "onChange"> & {
+    onChange?: (value: RPNInput.Value) => void
+  }
+
+const PhoneInput: React.ForwardRefExoticComponent<PhoneInputProps> =
+  React.forwardRef<React.ElementRef<typeof RPNInput.default>, PhoneInputProps>(
+    ({ className, onChange, ...props }, ref) => (
+      <RPNInput.default
+        ref={ref}
+        className={cn("flex", className)}
+        flagComponent={FlagComponent}
+        countrySelectComponent={CountrySelect}
+        inputComponent={InputComponent}
+        /**
+         * Handles the onChange event.
+         *
+         * react-phone-number-input might trigger the onChange event as undefined
+         * when a valid phone number is not entered. To prevent this,
+         * the value is coerced to an empty string.
+         *
+         * @param {E164Number | undefined} value - The entered value
+         */
+        onChange={(value) => onChange?.(value || "")}
+        {...props}
+      />
+    )
+  )
+PhoneInput.displayName = "PhoneInput"
+
+const InputComponent = React.forwardRef<HTMLInputElement, InputProps>(
+  ({ className, ...props }, ref) => (
+    <Input
+      className={cn("rounded-s-none rounded-e-lg", className)}
+      {...props}
+      ref={ref}
+    />
+  )
+)
+InputComponent.displayName = "InputComponent"
+
+type CountrySelectOption = { label: string; value: RPNInput.Country }
+
+type CountrySelectProps = {
+  disabled?: boolean
+  value: RPNInput.Country
+  onChange: (value: RPNInput.Country) => void
+  options: CountrySelectOption[]
+}
+
+const CountrySelect = ({
+  disabled,
+  value,
+  onChange,
+  options,
+}: CountrySelectProps) => {
+  const handleSelect = React.useCallback(
+    (country: RPNInput.Country) => {
+      onChange(country)
+    },
+    [onChange]
+  )
+
+  return (
+    <Popover>
+      <PopoverTrigger asChild>
+        <Button
+          type="button"
+          variant={"outline"}
+          className={cn("flex gap-1 rounded-e-none rounded-s-lg px-3")}
+          disabled={disabled}
+        >
+          <FlagComponent country={value} countryName={value} />
+          <ChevronsUpDown
+            className={cn(
+              "h-4 w-4 opacity-50 -mr-2",
+              disabled ? "hidden" : "opacity-100"
+            )}
+          />
+        </Button>
+      </PopoverTrigger>
+      <PopoverContent className="p-0 w-[300px]">
+        <Command>
+          <CommandList>
+            <CommandInput placeholder="Search country..." />
+            <CommandEmpty>No country found.</CommandEmpty>
+            <CommandGroup>
+              {options.map((option) => (
+                <CommandItem
+                  className="gap-2"
+                  key={option.value}
```

Comment:
Something is not right with the values as I get a warning about duplicate keys:

![React Warning About Duplicate Keys](https://github.com/shadcn-ui/ui/assets/64073162/dac41103-b7ef-4dbe-9faf-12d9d0809eb1)

```
Warning: Each child in a list should have a unique "key" prop. See https://reactjs.org/link/warning-keys for more information.
    at eval (webpack-internal:///(ssr)/./components/ui/command.tsx:128:84)
    at CountrySelect (webpack-internal:///(ssr)/./components/ui/custom/phone-input.tsx:76:26)
    at div
    at PhoneNumberInput_ (webpack-internal:///(ssr)/./node_modules/react-phone-number-input/modules/PhoneInputWithCountry.js:243:9)
    at eval (webpack-internal:///(ssr)/./node_modules/react-phone-number-input/modules/PhoneInputWithCountryDefault.js:62:34)
    at div
    at eval (webpack-internal:///(ssr)/./components/ui/custom/phone-input.tsx:27:83)
    at eval (webpack-internal:///(ssr)/./node_modules/@radix-ui/react-slot/dist/index.mjs:42:23)
    at eval (webpack-internal:///(ssr)/./node_modules/@radix-ui/react-slot/dist/index.mjs:16:23)
    at eval (webpack-internal:///(ssr)/./components/ui/form.tsx:100:87)
    at div
    at eval (webpack-internal:///(ssr)/./components/ui/form.tsx:65:84)
    at Controller (webpack-internal:///(ssr)/./node_modules/react-hook-form/dist/index.esm.mjs:511:39)
    at FormField (webpack-internal:///(ssr)/./components/ui/form.tsx:28:25)
    at div
    at div
    at eval (webpack-internal:///(ssr)/./components/ui/card.tsx:58:84)
    at div
    at eval (webpack-internal:///(ssr)/./components/ui/card.tsx:18:77)
    at div
    ...
```

I fixed it for now by changing the key to `key={option.label + option.value}`, although I don't like that method.

Reply:
"ZZ" means "International". And then as the solution; `key={option.value || "ZZ"}`.

Thank you for the review.  😊

Reply:
@NebraskaCoder 

"ZZ" means "International". And then as the solution; `key={option.value || "ZZ"}`.

Thank you for the review.  😊